server/client.go

```diff
@@ -2249,13 +2327,20 @@ func (c *client) msgHeader(mh []byte, sub *subscription, reply []byte) []byte {
 
 func (c *client) stalledWait(producer *client) {
 	stall := c.out.stc
+	ttl := 100 * time.Millisecond
+	if c.out.pb >= c.out.mp {
+		ttl = time.Second
```

Comment:
That feels like a long time..

Reply:
Since we don't "slow consumer" the routes/gws/ln, I wanted to put enough back pressure on the clients to avoid memory build-up. We can always revisit, but I think for now it is ok.

Reply:
Agree regarding revisit. I am working in JetStream branch on limits of memory etc and have a decent way I do that, we can do something similar here where we have a spike memory budget that once exceeded we will stall and release when it drops below.